crates/ruff_python_formatter/src/string/mod.rs

```diff
@@ -291,6 +291,30 @@ impl StringPart {
         }
     }
 
+    /// Returns the prefix of the string part.
+    pub(crate) const fn prefix(&self) -> StringPrefix {
+        self.prefix
+    }
+
+    /// Returns the surrounding quotes of the string part.
+    pub(crate) const fn quotes(&self) -> StringQuotes {
+        self.quotes
+    }
+
+    /// Returns the range of the string's content in the source (minus prefix and quotes).
+    pub(crate) const fn content_range(&self) -> TextRange {
+        self.content_range
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct StringNormalizer(StringPart);
+
```

Comment:
I think I would rather store `quoting`, `configured_style`, `parent_docstring_quote_char`, and `normalize_hex` rather than the part in the normalizer and expose a single `normalize(&self, part: &StringPart) -> NormalizedString` function. 

You could even have a single `StringNormalizer::from_context` method with methods like `with_prefered_quote_style` to configure it (builder like API).

This way the normalizer could be called multiple times, making it more independent from the `StringPart`. 

another alternative would be to add a `NormalizedString::from_part` method and move the normalization there.

Reply:
Thanks for the suggestions! Let me take a look at it tomorrow morning though